### PR TITLE
Fix conv_fill_spec_pad.mlir

### PR DESCRIPTION
**Pull request by yzhang93**
This change will get rid of the warning `conv_fill_spec_pad.mlir:57:3: warning: op argument #0 is not consumed in the body but is marked as consumed`.

Also attempt to fix https://github.com/nod-ai/iree-amd-aie/issues/704.

### CODE REVIEW

**Review by newling — APPROVED** [review]
Thanks! Seems like the conv_fill_spec_pad problem is solved.

**Comment by yzhang93** [comment]
> Thanks! Seems like the conv_fill_spec_pad problem is solved.

Well, I still see some failures for this test https://github.com/nod-ai/iree-amd-aie/actions/runs/10567875494/job/29277762727?pr=706#step:9:11037. But then it passed after repeating. So I guess now it doesn't consistently fail but only flaky.

**Comment by newling** [comment]
Ah ok I thought this one failed consistently on OSX and didn't have 5 chances to pass

File: tests/transform_dialect/conv_fill_spec_pad.mlir [hunk 1]
```diff
@@ -183,11 +183,11 @@ module attributes { transform.with_named_sequence } {
...
 
-    %conv_pre_contract =
-       transform.structured.match ops{["linalg.conv_1d_ncw_fcw"]}
-               in %fused_for_all : (!any) -> !any
+    // %conv_pre_contract =
+    //    transform.structured.match ops{["linalg.conv_1d_ncw_fcw"]}
+    //            in %fused_for_all : (!any) -> !any
 
-    transform.structured.vectorize %conv_pre_contract : !any
+    // transform.structured.vectorize %conv_pre_contract : !any
```
Comment (makslevental) [inline tests/transform_dialect/conv_fill_spec_pad.mlir:190]: Any reason to save this as a comment rather than just remove?

**Review by makslevental — APPROVED** [review]
Not familiar with these transform schedules but looks fine to me.

**Comment by makslevental** [comment]
Why not just get rid of it? Personally I'm not a fan of transform schedules but that's just me :shrug:

File: tests/transform_dialect/conv_fill_spec_pad.mlir (hunk 1 above)
Comment (yzhang93) [reply tests/transform_dialect/conv_fill_spec_pad.mlir:190]: Uh, I was just trying to see if this part of the codes caused flaky crash, but seems not. Without these, it still have some failures. So I'm going to keep blocking this test for MacOS Ci until figuring out the reason. 

Is there a MacOS machine that I can use to debug the CI failure?

**Pull request merged**